habitat-lab/habitat/config/CONFIG_KEYS.md

```diff
@@ -0,0 +1,180 @@
+
+
+
+
+## Environment
+Some habitat environment configurations.
+| Key | Description |
+| --- | --- |
+|habitat.environment.max_episode_steps| The maximum number of environment steps before the episode ends.|
+|habitat.environment.max_episode_seconds| The maximum number of seconds steps before the episode ends.|
+
+## Navigation Actions
+The way one would add an action to a configuration file would be by adding to the `defaults` list. For example:
+```
+defaults:
+  - /habitat/task/actions:
+    - move_forward
+    - turn_left
+```
+
+| Key | Description |
+| --- | --- |
+| habitat.task.actions.stop |     In Navigation tasks only, the stop action is a discrete action. When called, the Agent will request to stop the navigation task. Note that this action is needed to succeed in a Navigation task since the Success is determined by the Agent calling the stop action within range of the target. Note that this is different from the RearrangeStopActionConfig that works for Rearrangement tasks only instead of the Navigation tasks.|
+| habitat.task.actions.empty | In Navigation tasks only, the pass action. The robot will do nothing.|
+| habitat.task.actions.move_forward |     In Navigation tasks only, this discrete action will move the robot forward by a fixed amount determined by the `habitat.simulator.forward_step_size` amount. |
+| habitat.task.actions.turn_left |     In Navigation tasks only, this discrete action will rotate the robot to the left  by a fixed amount determined by the `habitat.simulator.turn_angle` amount. |
+| habitat.task.actions.turn_right |     In Navigation tasks only, this discrete action will rotate the robot to the right by a fixed amount determined by the `habitat.simulator.turn_angle` amount. |
+| habitat.task.actions.look_up |      In Navigation tasks only, this discrete action will rotate the robot's camera up by a fixed amount determined by the `habitat.simulator.tilt_angle` amount. |
+| habitat.task.actions.look_down |      In Navigation tasks only, this discrete action will rotate the robot's camera down by a fixed amount determined by the `habitat.simulator.tilt_angle` amount. |
+
+## Navigation Measures
+The way one would add an action to a configuration file would be by adding to the `defaults` list. For example:
+```
+defaults:
+  - /habitat/task/measurements:
+    - robot_force
+    - force_terminate
+```
+
+| Key | Description |
+| --- | --- |
+| habitat.task.measurements.num_steps| In both Navigation and Rearrangement tasks, counts the number of steps since  the start of the episode.|
+| habitat.task.measurements.distance_to_goal | In Navigation tasks only, measures the geodesic distance to the goal.|
+|habitat.task.measurements.distance_to_goal.distance_to | If 'POINT' measures the distance to the closest episode goal. If 'VIEW_POINTS' measures the distance to the episode's goal's viewpoint. |
```

Comment:
sorry I don't understand this one. if you could explain it to me, maybe we can re-write it?

Reply:
Sure. This is the code : 
```
            if self._distance_to == "POINT":
                distance_to_target = self._sim.geodesic_distance(
                    current_position,
                    [goal.position for goal in episode.goals],
                    episode,
                )
            elif self._distance_to == "VIEW_POINTS":
                distance_to_target = self._sim.geodesic_distance(
                    current_position, self._episode_view_points, episode
                )
            else:
                logger.error(
                    f"Non valid distance_to parameter was provided: {self._distance_to }"
                )

```
so if this config key us set to POINT, the distance is relative to the closest goal position. if it is VIEW_POINT, then the distance is relative to the closest view point. I **ASSUME** this is the difference between navigating close to an object and navigating to the POV from which an image of the object is taken.